### PR TITLE
Add idxmin() and idxmax()

### PR DESCRIPTION
**Pull request by Imperator26**
These are two procedures that return the index of the corresponding minimum and maximum value

### CODE REVIEW

File: lib/pure/httpclient.nim [hunk 1]
```diff
@@ -390,6 +390,7 @@ proc `[]=`*(p: var MultipartData, name: string,
 
 proc format(p: MultipartData): tuple[contentType, body: string] =
   if p == nil or p.content.len == 0:
+
```
Comment (cooldome) [inline lib/pure/httpclient.nim]: unrelated change please remove from this PR

**Comment by cooldome** [comment]
The aim is to keep system.nim lean. 
sequtils module is more suitable for these functions.

**Comment by juancarlospaco** [comment]
This should be moved to `sequtils`.
The return type should be `Natural`, then you can remove `result = 0`.

**Comment by cooldome** [comment]
Yep, return type is incorrect. Plus tests are required.

**Comment by SolitudeSF** [comment]
this doesn't work for arrays with non-zero starting index.

**Comment by disruptek** [comment]
I think this is too trivial and not useful enough to jump into stdlib.  It would need a `.since.` and a ChangeLog entry, also.
